extension/src/vscode/quickPick.ts

```diff
@@ -219,6 +220,55 @@ export const quickPickLimitedValues = <T>(
     quickPick.show()
   })
 
+const getUserOrderedSelection = <T>(
+  orderedSelection: T[],
+  newSelection: readonly QuickPickItemWithValue<T>[]
+) => {
+  const itemValues: T[] = []
+  for (const item of newSelection) {
+    if (!orderedSelection.includes(item.value)) {
+      orderedSelection.push(item.value)
+    }
+    itemValues.push(item.value)
+  }
+  return orderedSelection.filter(item =>
+    itemValues.some(val => isEqual(item, val))
+  )
+}
+
+export const quickPickUserOrderedValues = <T>(
```

Comment:
How could I test this function? Looked into creating a vscode test but can't figure out a way to trigger `onDidChangeSelection` in the test 🤔

Reply:
Figured it out! Took way to long to find the `workbench.action.quickPickManyToggle` command :woman_facepalming: